src/modules/ims_registrar_scscf/registrar_notify.c

```diff
@@ -2489,15 +2607,18 @@ void send_notification(reg_notification *n)
 
 
 		set_uac_req(&uac_r, &method, &h, 0, td, TMCB_LOCAL_COMPLETED,
-				uac_request_cb, 0);
+				uac_request_cb, watcher_contact);
 		tmb.t_request_within(&uac_r);
 	}
-	if(h.s)
-		pkg_free(h.s);
-	if(content.s) {
-		pkg_free(content.s);
+	watcher_contact = 0;
```

Comment:
isnt this leaking memory here?

Reply:
Shouldn't - both branches of the `if` before have the watcher_contact saved on the transaction. I think that freeing would result in a double-free, so instead we leave it for the `uac_request_cb` to clean up after the response or timeout is reached.